### PR TITLE
[tests] Add reconnect/backoff coverage for telemetry streaming event loops

### LINKED ISSUE #699: [tests] Add reconnect/backoff coverage for telemetry streaming event loops
## Why / Context
Reconnect behavior in telemetry loops drives runtime stability but can regress quietly. Deterministic backoff tests catch event-loop retry drift before production impact.

## Scope
**In scope**
- Add deterministic reconnect/backoff progression tests.
- Assert retry counter resets and gap bookkeeping on successful reconnect.
- Verify exhaustion paths produce actionable diagnostics.

**Out of scope**
- Replacing transport implementations.
- Integration tests requiring real sockets.

Constraints:
- Keep changes small/mergeable.
- Deterministic tests only.
- Avoid touching unrelated generated artifacts unless required.

## Acceptance Criteria (required)
- [ ] Reconnect/backoff timing behavior is deterministic under test clocks.
- [ ] Gap/retry counters are validated across failure and recovery paths.
- [ ] Retry exhaustion diagnostics are stable and asserted.

## Implementation Notes / Pointers
**Likely files / modules:**
- `src/gpt_trader/features/live_trade/engines/telemetry_streaming.py`
- `tests/unit/gpt_trader/features/live_trade/engines/test_telemetry_streaming.py`

**Related tests:**
- `tests/unit/gpt_trader/features/live_trade/engines/test_telemetry_streaming.py`
- `tests/unit/gpt_trader/monitoring/test_runtime_guards_base.py`

**Edge cases to handle:**
- Backoff ceilings should cap deterministically under repeated failures.
- Reconnect success should clear stale timers and retry state.
- Fixture clock skew should not introduce flaky assertions.

## Commands (local)
- `make fmt`
- `make lint`
- `uv run pytest -q tests/unit/gpt_trader/features/live_trade/engines/test_telemetry_streaming.py tests/unit/gpt_trader/monitoring/test_runtime_guards_base.py`

## PR Requirements
- PR title should match the issue.
- PR body must include: `Fixes #<issue-number>`
- CI must be green; if Agent Artifacts Freshness fails, run `uv run agent-regenerate`, then commit `var/agents/...` updates.

## Codex-Ready Checklist (for the issue creator)
- [x] Clear acceptance criteria
- [x] At least one file pointer
- [x] Commands included
- [x] No ambiguous "do the right thing" language

### PR DESCRIPTION
**Pull request by Solders-Girdles**
## Summary
- add reconnect/backoff regression coverage for telemetry streaming loops
- harden telemetry streaming reconnect behavior under transient failures
- verify loop lifecycle behavior across reconnect attempts in unit tests

Closes #699

### CODE REVIEW

File: src/gpt_trader/features/live_trade/engines/telemetry_streaming.py [hunk 1]
```diff
@@ -319,123 +389,154 @@ def _run_stream_loop(
...
+                    ctx.bot_id,
+                    {
+                        "event_type": WS_STREAM_RETRY_EVENT,
+                        "delay_seconds": delay,
+                        "error": str(exc),
+                        "gap_count": gap_count,
+                        "attempts": retry_state.attempts,
+                    },
+                )
+                time.sleep(delay)
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/live_trade/engines/telemetry_streaming.py]: Fixed in 93813897. Retry backoff now goes through `_sleep_with_stop_signal`, which uses `stop_signal.wait(delay)` when available so stop/restart can break backoff immediately instead of waiting the full sleep window. Added regression coverage in `tests/unit/gpt_trader/features/live_trade/engines/test_telemetry_streaming_retry_state.py::test_retry_backoff_wait_is_interruptible`.

File: src/gpt_trader/features/live_trade/engines/telemetry_streaming.py [hunk 2]
```diff
@@ -319,123 +389,154 @@ def _run_stream_loop(
...
-            stream = broker.stream_trades(symbols)
-        except Exception as trade_exc:
-            logger.error(
-                f"Failed to start streaming trades ({trade_exc})",
-                error=str(trade_exc),
-                operation="telemetry_stream",
-                stage="trades",
-            )
-            stream = None
+    retry_state = _get_retry_state(coordinator)
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/live_trade/engines/telemetry_streaming.py]: Fixed in 93813897. `_run_stream_loop` now resets `coordinator._stream_retry_state` at loop start, so a previous exhausted run cannot carry maxed attempts into the next restart. Added regression coverage in `tests/unit/gpt_trader/features/live_trade/engines/test_telemetry_streaming_retry_state.py::test_retry_state_resets_when_loop_restarts`.
